lib/puppet/provider/network_config/interfaces.rb

```diff
@@ -292,10 +289,24 @@ def self.format_file(filename, providers)
         stanza << "#{section} #{provider.send property}" if provider.send(property) and provider.send(property) != :absent
       end
 
+      if provider.mode == :vlan
+          if provider.options["vlan-raw-device"]
+            stanza << "vlan-raw-device #{provider.options["vlan-raw-device"]}"
+          else
+            vlan_range_regex = %r[\d{1,3}|40[0-9][0-5]]
```

Comment:
I assume your fork has that updated regex then, but the rest remains the same? If we take it in now, it would be nice to know it matches your fork, since you've given that a field test.